### DISCUSSION TITLE
PowerShell Script to copy files and folders

### DISCUSSION THREAD

**Discussion by Lascopee** [PowerShell/PowerShell#21175] [Q&A]
PowerShell Script to move Files, folders and sub folders from shared drive to SharePoint online maintaining the folder structure and last modified dates in from the Shared drive. I have attached the error message, any help to correct the script or a new a new script to execute the process will be appreciated. Thanks!
 
![github copy error](https://github.com/PowerShell/PowerShell/assets/158554786/ad7f6ae3-e9bd-4ecb-872a-78bb9a15a783)

**Comment by NoralK** [comment]
@Lascopee This is not the repository to ask this question, you want [PnP.PowerShell](https://github.com/pnp/powershell). Also, instead of the image of code it would have been better to paste the text of the code.

However, to answer your question, you are putting the name of the file in the folder of Add-PnPFile.

This code works as expected:

```
# Connect to SharePoint Online site
$siteURL = "https://tenant.sharepoint.com/sites/XYZ"
Connect-PnPOnline -Url $siteURL -Interactive

# Define the local path and the destination library name
$localFolderPath = "C:\Test"
$libraryName = "Archive"

# Recursively get all the file in the local folder path
$files = Get-ChildItem -Path $localFolderPath -Recurse -File

foreach($file in $files) {
    $filePath = $file.FullName.TrimEnd($file.Name)
    $relativePath = $filePath.Substring($localFolderPath.Length).replace("\","/").TrimStart("/")
    $serverRelativeFolder = "$libraryName/$relativePath"

    #ensure the target SharePoint folder exists
    $folderExist = Resolve-PnPFolder -SiteRelativePath $serverRelativeFolder

    try {
        $uploadResult = Add-PnPFile -Path $file.FullName -Folder $serverRelativeFolder -ErrorAction Stop
        Write-Host "Uploaded: $($file.FullName) to SharePoint"

        # Update file properties here
    } catch {
        Write-Host "Failed to upload: $($file.FullName). Error: $_" -ForegroundColor Red
    }

}

# Disconnect the session
```

Cheers,
Noral
